x/claims/keeper/abci.go

```diff
@@ -89,9 +89,13 @@ func (k Keeper) ClawbackEmptyAccounts(ctx sdk.Context, claimsDenom string) {
 	accPruned := int64(0)
 	accClawbacked := int64(0)
 
+	var addresses []sdk.AccAddress
+
 	k.IterateClaimsRecords(ctx, func(addr sdk.AccAddress, _ types.ClaimsRecord) (stop bool) {
-		// delete claims record once the account balance is clawed back
-		defer k.DeleteClaimsRecord(ctx, addr)
+		// NOTE: we cannot delete the record while iterating over it
+		defer func() {
```

Comment:
@fedekunze does this `defer` get executed at the end of the iteration or at the end of the whole method?

Reply:
updated the logic since the claim records shouldn't be deleted while iterating over the store items